src/mam/mod_mam_riak_timed_arch_yz.erl

```diff
@@ -347,11 +347,14 @@ get_mam_muc_gdpr_data(Username, Host) ->
     Jid = jid:make({LUser, LServer, <<>>}),
     LookupParams = ?DUMMY_LOOKUP_PARAMETERS#{with_jid := Jid},
     {ok, {_, _, Messages}} = lookup_messages([], Host, LookupParams),
-    Filtered = lists:filter(fun(El) -> is_muclight_message(Jid, El) end, Messages),
+    Filtered = lists:filter(fun({_,_,Msg}) -> is_groupchat_message(Msg) end, Messages),
```

Comment:
Previously, `lookup_messages` is extracting all messages that can be filtered with `with_jid := Jid` in the parameters. But these messages can be muc messages, and also private messages. We were previously matching on the nickname (that is what muclight is extracting, if the resource matches the jid, it is a muclight message, but then it wouldn't match classic mucs. What they both have in common, against private message, is the type, not the jids.